### PR TITLE
Redux in React

### PR DESCRIPTION
**Pull request by sentayhu19**
**In this pull req**

- `State` is immutable in reducers used
- The default state in  books reducer from an empty array to an array with a few books. is changed
- App connected to the Redux store.
- Displays books from the Redux store
- `Useselectors` to consume the state used
- Functionality for "Add new" button added
- Set the data inputs in the local React state.
- Functionality for "Remove" button added
- `Dispatch` a corresponding action 
- Functionality for "Check status" button added
- Import Redux `useDispatch`.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
+  id: generate({ charset: alphanumeric }),
+  title: 'The book of boba fet',
+  author: 'Lucas',
+  completed: false,
+},
+{
+  id: generate({ charset: alphanumeric }),
+  title: 'The mandalorian',
+  author: 'JJ',
+  completed: false,
```
Comment (dasileker) [inline src/redux/books/books.js]: - [x] Please remove all the hard-coded objects that you have added to display the book.

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
+  'Economy',
+  'Fantasy',
+  'Historical Fiction',
+  'Horror',
+  'Literary Fiction',
+  'Romance',
+  'Short Stories',
+  'Suspense and Thrillers',
+  'Sciences Fiction',
+  'True Crime',
```
Comment (dasileker) [inline src/redux/categories/categories.js]: - [x] The categories are not showing in the add books, so no need for them right now.

File: src/components/Categories/Categories.js [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
-const Categories = () => (
-  <>
-    <button className="Check-status" type="button">Check status</button>
-  </>
-);
+const Categories = () => {
+  const dispatch = useDispatch();
+  return (
+    <>
+      <button className="Check-status" onClick={() => dispatch(checkStatus())} type="button">Check status</button>
```
Comment (dasileker) [inline src/components/Categories/Categories.js]: - [x] Please make the dispatch for the `check status` when the button is clicked to display the under construction, right now the `Onclick` is empty.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @sentayhu19,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Linters are good ✅.
- PR is nice ✅.

## Required Changes ♻️

_Check the comments under the review._


- Hello and nice work on the Book store however, there is some minor issue that you need to take care of. 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- **_Please ensure that the readme file is having a `Getting started with full directions from copying the repo to opening the live server._**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi @sentayhu19 🙋‍♀️,  

Great work! 💯 
You have done a great job implementing the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

# Status: Approved ✔️ 

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**